data_structures/hash_table/isomorphic_strings.rb

```diff
@@ -0,0 +1,63 @@
+# Challenge name: Isomorphic Strings
+#
+# Given two strings s and t, determine if they are isomorphic.
+# Two strings s and t are isomorphic if the characters in s can be replaced to get t.
+# All occurrences of a character must be replaced with another character while preserving the order of characters. 
+# No two characters may map to the same character, but a character may map to itself.
+#
+# Example 1:
+# Input: s = "egg", t = "add"
+# Output: true
+#
+# Example 2:
+# Input: s = "foo", t = "bar"
+# Output: false
+#
+# Example 3:
+# Input: s = "paper", t = "title"
+# Output: true
+#
+# Constraints:
+# 1 <= s.length <= 5 * 104
+# t.length == s.length
+# s and t consist of any valid ascii character.
+
+# Approach 1: Hash Map
+# Time Complexity: O(N)
```

Comment:
What's the space complexity of this algorithm?

Reply:
> What's the space complexity of this algorithm?

Would it be O(N) because of the loop?

Reply:
In this case, I believe you'd have a space complexity equal to N + M where N is the size of the `s` and M the size of the `r` array.

Reply:
I think you're right. It's`O(N)` where N is the length of the `s` array.